src/basic/format-util.c

```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "format-util.h"
+#include "memory-util.h"
+
+char *format_ifname(int ifindex, char buf[static IF_NAMESIZE + 1]) {
+        /* Buffer is always cleared */
+        memzero(buf, IF_NAMESIZE + 1);
+        return if_indextoname(ifindex, buf);
```

Comment:
I looked into this a bit. `if_indextoname` indeed does no termination itself, and the documentation says nothing beyond the required size of the buffer. But the kernel seems to do termination on its own, in `dev_ifname`, and the return value from the SIOCGIFNAME ioctl is passed straight through to userspace. So I guess we could rely on this internal implementation detail of the kernel, and create a buffer of size IF_NAMESIZE and maybe add an assert here that the last byte is indeed 0.

@poettering ?

Reply:
OK, let's merge. @poettering most likely will not have time to review this soon anyway. It makes sense to merge this, and maybe we can add a patch to use a smaller buffer later. It'll be much easier, because *this* patch cleans up the buffer sizes and makes them consistent.